src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,153 @@
+     );
+
+    public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            string id = diagnostic.Id;
+            switch (id)
+            {
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Logging properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Logging")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id),
+                            diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureServices.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Services properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Services")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DoNotUseHostConfigureServices.Id),
+                            diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id:
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            "Fix references to Configuration properties on WebApplicationBuilder",
+                            cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, SyntaxFactory.IdentifierName("Configuration")),
+                            equivalenceKey:
+                            DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id),
+                            diagnostic);
+                    break;
+            }
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static async Task<Document> FixWebApplicationBuilder(Diagnostic diagnostic, Document document, CancellationToken cancellationToken, IdentifierNameSyntax identifierMethod)
+    {
+        var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root == null)
+        {
+            return document;
+        }
+
+        // builder.Host.ConfigureLogging(builder => builder.AddJsonConsole());
+        var diagnosticTarget = root.FindNode(diagnostic.Location.SourceSpan); 
```

Comment:
Since `ConfigureLogging` doesn't return null, the expression `builder.Host.ConfigureLogging(builder => builder.AddJsonConsole())` might be an argument to another method right?

In that case `FindNode` will return `ArgumentSyntax` instead of `InvocationExpressionSyntax`.
To correctly get `InvocationExpressionSyntax` you care about you need to pass `getInnermostNodeForTie: true` to `FindNode`.

Consider trying to craft a test where the fixer will fail without `getInnermostNodeForTie: true`.